extensions/cornerstone/src/components/WindowLevelActionMenu/WindowLevelActionMenu.tsx

```diff
@@ -59,6 +99,94 @@ export function WindowLevelActionMenu({
     [commandsManager]
   );
 
+  const onSetColorLUT = useCallback(
+    props => {
+      const immediate = true;
+      commandsManager.run({
+        commandName: 'setViewportColormap',
+        commandOptions: {
+          ...props,
+          immediate,
+        },
+        context: 'CORNERSTONE',
+      });
+    },
+    [commandsManager]
+  );
+
+  const onSetColorbar = useCallback(
+    props => {
+      const currentColorMap = getCurrentViewportColormap(viewportId);
+      // incase an initialColormap for the colorbar is set, but they have no colormap on the viewport
+      // if they have a colormap on the viewport, that takes priority
+      if (props.options.activeColormapName && !currentColorMap) {
+        onSetColorLUT({
+          viewportId,
+          colormap: colormaps.find(c => c.Name === props.options.activeColormapName),
+        });
+      }
+      commandsManager.run({
+        commandName: 'toggleViewportColorbar',
+        commandOptions: {
+          ...props,
+        },
+        context: 'CORNERSTONE',
+      });
+    },
+    [commandsManager]
+  );
+
+  const getCurrentViewportColormap = viewportId => {
+    const { properties } = cornerstoneViewportService.getPresentation(viewportId);
+    const colormap = properties?.colormap;
+    if (!colormap) {
+      return colormaps.find(c => c.Name === 'Grayscale') || colormaps[0];
+    }
+    return colormap;
+  };
+
+  useEffect(() => {
+    const updateColorbarState = () => {
+      setShowColorbar(colorbarService.isColorbarToggled(viewportId));
+    };
+
+    const { unsubscribe } = colorbarService.subscribe(
+      colorbarService.EVENTS.COLORBAR_STATE_CHANGED,
+      updateColorbarState
+    );
+
+    return () => {
+      unsubscribe();
+    };
+  }, []);
+
+  useEffect(() => {
+    const colorbarState = colorbarService.getColorbarState(viewportId);
+    if (colorbarState && colorbarState.needsRefresh) {
+      window.setTimeout(() => {
+        colorbarService.removeColorbar(viewportId);
+        onSetColorbar({
+          viewportId,
+          options: {
+            colormaps,
+            ticks: {
+              position: colorbarTickPosition,
+            },
+            width: colorbarWidth,
+            position: colorbarContainerPosition,
+            activeColormapName: colorbarInitialColormap,
+          },
+        });
+      }, 0);
+    }
+    return () => {
+      const colorbarState = colorbarService.getColorbarState(viewportId);
+      if (colorbarState) {
+        colorbarService.markForRefresh(viewportId, true);
+      }
```

Comment:
This is fine for one line indentation, but if you see above it is 10+ lines

Reply:
done